llama.cpp

```diff
@@ -12771,8 +12773,10 @@ void llama_grammar_accept_token(struct llama_context * ctx, struct llama_grammar
     // Note terminating 0 in decoded string
     const auto   decoded     = decode_utf8(piece, grammar->partial_utf8);
     const auto & code_points = decoded.first;
+    std::vector<std::vector<const llama_grammar_element *>> tmp_new_stacks;
     for (auto it = code_points.begin(), end = code_points.end() - 1; it != end; ++it) {
-        grammar->stacks = llama_grammar_accept(grammar->rules, grammar->stacks, *it);
+        llama_grammar_accept(grammar->rules, grammar->stacks, *it, tmp_new_stacks);
+        tmp_new_stacks.swap(grammar->stacks);
```

Comment:
```suggestion
        grammar->stacks = tmp_new_stacks.swap;
```

Reply:
I know this is an old comment on an old PR, but I have since learned that my comment here was invalid -- it appears that .swap() is faster than assignment:
https://stackoverflow.com/questions/22001854/faster-to-swap-or-assign-a-vector-of-strings

I don't expect anything to be done about it now, but more making a note to myself to keep it in mind for the future!

Reply:
Yes swap is instant, but readability matters and I checked with and without it in this specific instance and don’t think it impacted overall performance